canvas_exercise/shapes_game/index.js

```diff
@@ -1,36 +1,94 @@
 window.addEventListener("load", function() {
+  var canvas = document.getElementById("shapes-game"),
+    height = canvas.scrollHeight,
+    width = canvas.scrollWidth,
+    gameOn = false,
+    expectedKey = undefined,
+    ctx = canvas.getContext('2d'),
+    // white triangle = up, red square = down,
+    // red triangle = left, white square = right
+    expectedKeysMap = {white0: 38, red1: 40, red0: 37, white1: 39},
+    timerSpan = document.getElementById("time-remaining"),
+    scoreSpan = document.getElementById("score-val"),
+    seconds = 3,
+    intervalId;
+
+  canvas.width = width;
+  canvas.height = height;
 
-  function clear(ctx, width, heigt) {
+  drawGameStartText()
+
+  function clear() {
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
   }
 
-  function drawRandomShape(ctx, width, height) {
+  function drawRandomShape() {
+    var randomShape = ['white0', 'red1', 'red0', 'white1'][Math.floor(Math.random()*4)],
+        x = Math.floor(Math.random()*(width-100)),
+        y = Math.floor(Math.random()*(height-100));
+        ctx.fillStyle = randomShape.slice(0, -1);
+
+    clear()
+
+    function square(){
+      ctx.fillRect(x,y,100,100);
+    }
+
+    function triangle(){
+      ctx.beginPath();
+      ctx.moveTo(x,y);
+      ctx.lineTo(100+x, 100+y);
+      ctx.lineTo(x, 100+y);
+      ctx.fill();
+      ctx.closePath();
+    }
+
+    if(+randomShape[randomShape.length-1] === 1){
+      square()
+    } else {
+      triangle()
+    }
+    expectedKey = randomShape;
   }
 
-  function drawGameStartText(ctx, width, height, score) {
+
+  function drawGameStartText() {
+    clear()
+    ctx.font = '30px serif';
+    ctx.fillStyle = 'white';
+    ctx.fillText('Press the space bar to start a new game', 150, 350);
   }
 
-  function restartGame(ctx, width, height) {
+  function restartGame() {
+    timerSpan.innerText = 30;
+    scoreSpan.innerText = 0;
+    gameOn = false;
+
+    drawGameStartText()
   }
 
-  var canvas = document.getElementById("shapes-game"),
-      height = canvas.scrollHeight,
-      width = canvas.scrollWidth,
-      gameOn = false,
-      expectedKey = undefined,
-      ctx = canvas.getContext('2d'),
-      // white triangle = up, red square = down,
-      // red triangle = left, white square = right
-      expectedKeysMap = {white0: 38, red1: 40, red0: 37, white1: 39},
-      timerSpan = document.getElementById("time-remaining"),
-      scoreSpan = document.getElementById("score-val"),
-      seconds = 3,
-      intervalId;
+  document.addEventListener("keyup", function(e) {
+    if(gameOn){
+      if(e.keyCode === expectedKeysMap[expectedKey]){
```

Comment:
There is a bug where my score gets decreased when I press a space bar. It might be related to this code